### PR TITLE
Refactor/UI test improvement

### LINKED ISSUE #78: Create a testing suite and refactor UI tests to use it
Create a testing suite to optimize test time by reducing the number of overal tests while still keeping the same coverage and usefullness as the refactored tests.
Apply this suite to the UI tests as they are the main bottleneck.

### LINKED ISSUE #95: Refactor tests 2
Refactor the UI tests to use the new report system to make them faster and more efficient.

### PR DESCRIPTION
**Pull request by dtbcds**
### Summary
This PR refactors nearly all existing UI tests, focusing on improving performance and maintainability without sacrificing coverage.

### Key Changes
- Added a Checkpoint utility for tests.
- Refactored most of the existing UI tests for speed improvement and better readability.  
- **More than doubled** the overall test execution speed.  
- Maintained full test coverage — with **additional coverage gained** in several areas.  
- Reduced redundant test logic and optimized setup/teardown operations.  
- Checkpoint system revolves around merging tests with similar setups into one big test that still provides error's and logs with the same pinpoint accuracy as many tests if not more.

### Impact
- Significantly faster CI runs.  
- More reliable and consistent test outcomes.  
- Easier to maintain and extend test suite in the future.  

### Results
- **~2x faster** total test execution time.  
- **Improved coverage** in edge cases and UI interaction flows.

### CODE REVIEW

File: app/src/androidTest/java/com/github/meeplemeet/ui/CreateAccountScreenTest.kt [hunk 1]
```diff
@@ -48,158 +51,155 @@ class CreateAccountScreenTest : FirestoreTests() {
...
+    checkpoint("Multiple clicks end with owner ON / renter OFF") {
+      /* 3 clicks instead of 5 – still exercising the UI */
+      ownerCheckbox().performClick()
+      val h3 = "h_" + UUID.randomUUID().toString().take(8).also { handlesCreated += it }
+      handleField().performTextInput(h3)
+      usernameField().performTextInput("Frank")
+      submitBtn().performClick()
+
+      compose.waitForIdle()
+      Thread.sleep(1000) // Give time for async operations
```
Comment (NilSamaha) [inline app/src/androidTest/java/com/github/meeplemeet/ui/CreateAccountScreenTest.kt]: If this sleep is needed, I would suggest to use compose.waitUntil(...) instead

File: app/src/androidTest/java/com/github/meeplemeet/ui/CreateAccountScreenTest.kt [hunk 2]
```diff
@@ -48,158 +51,155 @@ class CreateAccountScreenTest : FirestoreTests() {
...
+      usernameField().performTextClearance()
+
+      renterCheckbox().performClick()
+      val h4 = "h_" + UUID.randomUUID().toString().take(8).also { handlesCreated += it }
+      handleField().performTextInput(h4)
+      usernameField().performTextInput("Frank")
+      submitBtn().performClick()
+
+      compose.waitForIdle()
+      Thread.sleep(1000) // Give time for async operations
```
Comment (NilSamaha) [inline app/src/androidTest/java/com/github/meeplemeet/ui/CreateAccountScreenTest.kt]: If this sleep is needed, I would suggest to use compose.waitUntil(...) instead

File: app/src/androidTest/java/com/github/meeplemeet/ui/CreateAccountScreenTest.kt [hunk 3]
```diff
@@ -48,158 +51,155 @@ class CreateAccountScreenTest : FirestoreTests() {
...
+        renterCheckbox().performClick() // off
+        ownerCheckbox().performClick() // on
+
+        val h5 = "h_" + UUID.randomUUID().toString().take(8).also { handlesCreated += it }
+        handleField().performTextInput(h5)
+        usernameField().performTextInput("Frank")
+        submitBtn().performClick()
+
+        compose.waitForIdle()
+        Thread.sleep(1000) // Give time for async operations
```
Comment (NilSamaha) [inline app/src/androidTest/java/com/github/meeplemeet/ui/CreateAccountScreenTest.kt]: If this sleep is needed, I would suggest to use compose.waitUntil(...) instead

File: app/src/androidTest/java/com/github/meeplemeet/ui/CreateSessionScreenTest.kt [hunk 4]
```diff
@@ -182,172 +165,139 @@ class CreateSessionScreenTest {
...
+  }
+
+  // Grouped UI: bars, snackbar, back/discard
+  @Test
+  fun full_smoke() {
+    val harness =
+        ComposeOnceHarness(account = me, startDisc = baseDiscussion, viewModel = createSessionVM)
+    compose.setContent { harness.Content() }
+
+    checkpoint("ui_chrome_present_and_nav_discard_callbacks_fire") {
```
Comment (NilSamaha) [inline app/src/androidTest/java/com/github/meeplemeet/ui/CreateSessionScreenTest.kt:195]: consider decomposing long checkpoints into two (to avoid long and broad names, that could be an issue when debugging)

File: app/src/androidTest/java/com/github/meeplemeet/ui/MapScreenTest.kt [hunk 5]
```diff
@@ -117,295 +121,278 @@ class MapScreenTest : FirestoreTests(), OnMapsSdkInitializedCallback {
...
 
-  @Test
-  fun fab_visibleForShopOwner() {
-    setContent(account = shopOwnerAccount)
-    compose.waitForIdle()
-    compose.onNodeWithTag(MapScreenTestTags.ADD_FAB).assertIsDisplayed()
-    compose.onNodeWithTag(MapScreenTestTags.ADD_FAB).assertHasClickAction()
-  }
+    // ========================================================================
+    // Test 1: Initial Map Display + Screen Elements
```
Comment (NilSamaha) [inline app/src/androidTest/java/com/github/meeplemeet/ui/MapScreenTest.kt:130]: I really liked these comments, it helps a lot visually. Since the restructure makes it harder to see the different test, consider adding it on other tests

File: app/src/androidTest/java/com/github/meeplemeet/ui/MapScreenTest.kt [hunk 6]
```diff
@@ -117,295 +121,278 @@ class MapScreenTest : FirestoreTests(), OnMapsSdkInitializedCallback {
...
-            owner = shopOwnerAccount,
-            name = "Test Board Game Shop",
-            address = testLocation,
-            openingHours = testOpeningHours)
-
-    setContent()
-    compose.waitForIdle()
+        val pin = state.geoPins.first()
+        mapViewModel.selectPin(pin)
+        delay(1000)
```
Comment (NilSamaha) [inline app/src/androidTest/java/com/github/meeplemeet/ui/MapScreenTest.kt:177]: remove delay if not needed and add comments if necessary (try to use compose.waitUntil if you can in these cases)

**Review by NilSamaha — COMMENTED** [review]
I really like how you structured the tests to minimize the waiting time when creating the composables.  
It makes the tests much more efficient and CI pass faster   

I also noticed that in some files, the comments are really well done and help visualize the different test sections more clearly.  
That’s especially helpful now that we’re structuring things around one big test per screen.  
If you could make the comments in the other tests consistent with that style, it would really improve readability.  

---

### A couple of small points:
- I saw some `Thread.sleep` calls — these could probably be replaced with `compose.waitUntil()` to make the tests more stable and faster.  
- For the delays inside `MapScreenTest`, it might be worth double-checking if they’re strictly necessary (I assume they might be there because of the map’s loading time).  
  The structure and comments in that file are great though — if you can replicate that approach in the other tests, that would be awesome.  
- For longer checkpoints, maybe try splitting them into smaller, more focused ones for clarity.  
- The PR is quite large overall — next time, it would be great to split it into two smaller PRs if possible. It makes reviewing and testing changes much easier.  

---

Overall, really nice work on improving the test structure and readability.

**Review by NilSamaha — APPROVED** [review]
I really like how you structured the tests to minimize the waiting time when creating the composables.  
It makes the tests much more efficient and CI pass faster   

I also noticed that in some files, the comments are really well done and help visualize the different test sections more clearly.  
That’s especially helpful now that we’re structuring things around one big test per screen.  
If you could make the comments in the other tests consistent with that style, it would really improve readability.  

---

### A couple of small points:
- I saw some `Thread.sleep` calls — these could probably be replaced with `compose.waitUntil()` to make the tests more stable and faster.  
- For the delays inside `MapScreenTest`, it might be worth double-checking if they’re strictly necessary (I assume they might be there because of the map’s loading time).  
  The structure and comments in that file are great though — if you can replicate that approach in the other tests, that would be awesome.  
- For longer checkpoints, maybe try splitting them into smaller, more focused ones for clarity.  
- The PR is quite large overall — next time, it would be great to split it into two smaller PRs if possible. It makes reviewing and testing changes much easier.  

---

Overall, really nice work on improving the test structure and readability.

File: app/src/androidTest/java/com/github/meeplemeet/ui/MapScreenTest.kt (hunk 6 above)
Comment (dtbcds) [reply app/src/androidTest/java/com/github/meeplemeet/ui/MapScreenTest.kt:177]: delay is needed, compose.waitUntil doesn't work here.

File: app/src/androidTest/java/com/github/meeplemeet/ui/CreateSessionScreenTest.kt (hunk 4 above)
Comment (dtbcds) [reply app/src/androidTest/java/com/github/meeplemeet/ui/CreateSessionScreenTest.kt:195]: These can't be decomposed since it is all sanity checks for a single state

**Review by NilSamaha — APPROVED** [review]
Great that you could remove the Thread.sleep, LGTM
